### DISCUSSION TITLE
Using own PLY data for Cluster Classification using CGAL

### DISCUSSION THREAD

**Discussion by pmff96** [CGAL/cgal#5416] [Q&A] [answered]
Hi! This question might seem very simple, I apologize in advance, but I'm having trouble understanding what I must do to perform cluster classification using CGAL on my own dataset, already in PLY format. Is it just as simple as changing the file directory and labels to correspond to my data or do I have to change anything else (like the bin file, for example)? In case I need to change the bin file, or generate my own, is there any explanation on how to do it? Once again, I apologize for my question, I'm fairly new at working with CGAL and PLY in general.

**Comment by sgiraudot** [comment, marked as answer]
Hello,

The bin file is a saved trained classifier, which is inherently dependent on what you want to classify: it will only work if you use the same set of labels and features as in the example. Even then, it's not likely to give good results unless your data is, by chance, geometrically similar to the data of the example.

The usual way to do classification is to provide a training set (a part of the data that you label by hand using whatever labels you want to classify) and to use the `train()` function of the classifier (see https://doc.cgal.org/latest/Classification/index.html#title26 ). Then, the trained classifier can be saved (in a bin file) and reused again later.

**Reply by pmff96** [reply]
Sorry for the late reply, so I know it is possible to manually label data using the polyhedron demo, and that it is possible to train and classify data in that same demo, but is it possible to save that bin file so I can then use it for cluster classification? The polyhedron demo works great, but for the purpose of my study I was trying to first make a segmentation process to obtain clusters and then classify those clusters, instead of classifying the whole scene.

**Reply by sgiraudot** [reply]
Yes, you can indeed use the demo. If your point cloud has identified clusters, the Classification plugin will ask if you want to use clusters as input instead of raw points. If you use your own program to generate clusters, these clusters should be stored as an `int` property called "shape" in your PLY input (assigning the index of the cluster to each point, with -1 for points not assigned to any cluster). Otherwise, you can use the shape detection plugin (RANSAC or Region Growing) in the demo, which will generate the right `int` "shape" property.

**Reply by pmff96** [reply]
Thank you very much for your help, I think I understand how the cluster classification algorithm works a bit better now
